From ff7158fe27c5f8c098e193fd47f54cc043638bdb Mon Sep 17 00:00:00 2001
From: Senthilkumar N L <snlakshm@codeaurora.org>
Date: Wed, 27 May 2015 12:11:19 +0800
Subject: ipq40xx: enable quad core bringup

Enable secondary cores reset out sequence

Signed-off-by: Senthilkumar N L <snlakshm@codeaurora.org>
(cherry picked from commit 173ec83661a1d68ef2883aa775822b0b5770a4a5)

Change-Id: I4daeae536ec5f68321902ca2d2f69fc3bffc7be7
Signed-off-by: Prasanna Kumar Thoorvas Samyrao Muralidharan <ptsm@codeaurora.org>
---
 arch/arm/mach-qcom/platsmp.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/arch/arm/mach-qcom/platsmp.c b/arch/arm/mach-qcom/platsmp.c
index f476a71..b363890d 100644
--- a/arch/arm/mach-qcom/platsmp.c
+++ b/arch/arm/mach-qcom/platsmp.c
@@ -199,23 +199,23 @@ static int a7ss_release_secondary(unsigned int cpu)
 	if (!base)
 		return -ENOMEM;
 
+	/* Enable Clamp signal and assert core reset */
+	writel_relaxed(0x00000033, base + 0x04);
+	mb(); /* barrier */
 
-	printk("Fix me, the below initializations should be done by rom ?");
+	/* Set GDHS and delay counter */
+	writel_relaxed(0x20000001, base + 0x14);
+	mb(); /* barrier */
 
-	if (0) {
-		writel_relaxed(0x00000033, base + 0x04);
-		mb(); /* barrier */
-
-		writel_relaxed(0x10000001, base + 0x14);
-		mb(); /* barrier */
-		udelay(2);
+	udelay(2);
 
-		writel_relaxed(0x00020008, base + 0x04);
-		mb(); /* barrier */
+	/* Enable Core memory HS */
+	writel_relaxed(0x00020008, base + 0x04);
+	mb(); /* barrier */
 
-		writel_relaxed(0x00020088, base + 0x04);
-		mb(); /* barrier */
-	}
+	/* Report that the CPU is powered up */
+	writel_relaxed(0x00020088, base + 0x04);
+	mb(); /* barrier */
 
 	iounmap(base);
 	return 0;
-- 
cgit v1.1
